src/app/dataset-view/additional-components/data-component/data.component.ts

```diff
@@ -111,10 +116,43 @@ export class DataComponent extends BaseComponent implements OnInit {
         }
     }
 
-    public addData(): void {
-        const modalRef: NgbModalRef = this.ngbModalService.open(AddDataModalComponent);
-        const modalRefInstance = modalRef.componentInstance as AddDataModalComponent;
-        modalRefInstance.datasetBasics = this.datasetBasics;
+    public addData(overviewUpdate: OverviewUpdate): void {
+        const metadata = overviewUpdate.overview.metadata;
+        if (metadata.currentPollingSource || metadata.currentPushSources.length || metadata.currentTransform) {
```

Comment:
How having push sources will help? You cannot launch a push-source dataset for update

Reply:
Yes, agree